domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -123,6 +132,33 @@ constructor(
 
     private lateinit var imageCaptureUseCase: ImageCapture
 
+    /**
+     * Applies a CaptureCallback to the provided image capture builder
+     */
+    @OptIn(ExperimentalCamera2Interop::class)
+    private fun onCaptureCompletedCallback(imageCaptureBuilder: ImageCapture.Builder) {
+        val captureCallback = object : CameraCaptureSession.CaptureCallback() {
+            override fun onCaptureCompleted(
+                session: CameraCaptureSession,
+                request: CaptureRequest,
+                result: TotalCaptureResult
+            ) {
+                super.onCaptureCompleted(session, request, result)
+                try {
+                    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.Q) {
+                        Trace.endAsyncSection(FIRST_FRAME_TRACE, FIRST_FRAME_COOKIE)
```

Comment:
If this is on Preview, you'll only want to call this once. Since it's in a callback, you could just have a boolean instance variable to check if it's already been called, and set it to `true` once you call it the first time.